pkg/transformers/imagetag.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformers
+
+import (
+	"strings"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/resmap"
+	"github.com/kubernetes-sigs/kustomize/pkg/types"
+)
+
+// imageTagTransformer replace image tags
+type imageTagTransformer struct {
+	imageTags []types.ImageTag
+}
+
+var _ Transformer = &imageTagTransformer{}
+
+// NewImageTagTransformer constructs a imageTagTransformer.
+func NewImageTagTransformer(slice []types.ImageTag) (Transformer, error) {
+	return &imageTagTransformer{slice}, nil
+}
+
+// Transform finds the matching images and replace the tag
+func (pt *imageTagTransformer) Transform(resources resmap.ResMap) error {
+	for _, res := range resources {
+		err := pt.replaceTag(res.UnstructuredContent())
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// replaceTag replaces the image tags inside one object
+// It loops though all images inside containers session, finds matched ones and update the tag name
+func (pt *imageTagTransformer) replaceTag(obj map[string]interface{}) error {
```

Comment:
please break this long method into shorter ones.  too much indentation

Reply:
I split it into two methods. See if it is clear and easy to understand.